consensus/hotstuff/notifications/telemetry.go

```diff
@@ -25,20 +25,25 @@ import (
 //     -- a local timeout has been initiated
 //   - Each path through the state machine is identified by a unique id.
 //
+// Additionally, the TelemetryConsumer reports events related to vote and timeout aggregation
+// but those events are not bound to a path, so they are reported differently.
 // Generally, the TelemetryConsumer could export the collected data to a variety of backends.
 // For now, we export the data to a logger.
 //
 // Telemetry does NOT capture slashing notifications
 type TelemetryConsumer struct {
 	NoopConsumer
-	pathHandler *PathHandler
+	pathHandler  *PathHandler
```

Comment:
+1 for option 2

Reply:
I assume option 2 would entail a bunch of changes across many files. Maybe it would be best to commit this PR first, adding just a TODO comment and then follow up with a PR for the interface refactoring.

Reply:
While I agree that our consumer is rather big at the moment and some decoupling would be beneficial I don't particularly agree that we have an issue right now. Yes, we have some white-spaces in documentation where we don't specifically mention that we it's intended for single-threaded usage but we implicitly guarantee it based on how we use it. All calls to `pathHandler` happen exclusively in `EventHandler` or code which is called by `EventHandler`, this is bound to notion of _path_. Since `EventHandler` opens a path and closes it, and events that use `pathHandler` must be part of _some_ path. For other events(vote and timeout aggregation) we specifically use `noPathHandler` which is a plain logger atm. 

I also thing that option 2 is better and cleaner, created an issue: https://github.com/dapperlabs/flow-go/issues/6453